main.go

```diff
@@ -189,36 +192,73 @@ func run(state overseer.State) {
 		if remote {
 			defer os.RemoveAll(repoPath)
 		}
-		err = e.ScanGit(ctx, repoPath, *gitScanBranch, *gitScanSinceCommit, *gitScanMaxDepth, filter)
-		if err != nil {
+
+		cfg = sources.Config{
+			RepoPath: repoPath,
+			HeadRef:  *gitScanBranch,
+			BaseRef:  *gitScanSinceCommit,
+			MaxDepth: *gitScanMaxDepth,
+			Filter:   filter,
+		}
+		if err = e.ScanGit(ctx, &cfg); err != nil {
```

Comment:
**Question/Suggestion:** What is the reasoning for using a pointer here for the `cfg`?

I would prefer passing the struct so there's no possiblity of null pointer dereference. Alternatively, if we need to pass a pointer, we should check for a nil config at the beginning of each of the scan functions.

Reply:
Will do.